### PR TITLE
libzip: install man-pages and add optionas for three optional dependencies

### PR DESCRIPTION
**Pull request by nagy**
###### Motivation for this change
This builds man-pages for `libzip` and adds three optional build arguments. Without this, some html manpages are installed, but I think it is better to install the regular man-pages instead.

The cmake files do not allow for both (man-pages and html-pages) to be installed without them being patched.

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/development/libraries/libzip/default.nix [hunk 1]
```diff
@@ -4,14 +4,21 @@
 , fetchurl
 , perl
 , zlib
+, groff
+, withBzip2 ? false
+, bzip2
+, withLZMA ? false
+, lzma
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libzip/default.nix:11]:
```suggestion
, xz
```

lzma is an alias to xz which is going to be removed soon.

File: pkgs/development/libraries/libzip/default.nix [hunk 2]
```diff
@@ -23,20 +30,23 @@ stdenv.mkDerivation rec {
...
   ];
 
-  outputs = [ "out" "dev" ];
+  outputs = [ "out" "dev" "man" ];
 
-  nativeBuildInputs = [ cmake perl ];
+  nativeBuildInputs = [ cmake perl groff ];
   propagatedBuildInputs = [ zlib ];
+  buildInputs = (lib.optional withLZMA lzma) ++ (lib.optional withBzip2 bzip2)
+    ++ (lib.optional withOpenssl openssl );
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libzip/default.nix:38]:
```suggestion
  buildInputs = lib.optionals withLZMA [ xz ]
    ++ lib.optionals withBzip2 [ bzip2 ]
    ++ lib.optionals withOpenssl [ openssl ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Do we want to enable them by default?

**Comment by nagy** [comment]
> Do we want to enable them by default?

According to ofborg, around 100 to 500 packages depend on `libzip`. This means we would increase the closure size for all these packages while non of them use these features. I think the best way for packages what need these features would be a local `.override`, but I am not against it. I just tend towards no.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118464` at f8df64c8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>56 packages marked as broken and skipped:</summary>
  <ul>
    <li>fileshelter</li>
    <li>gimpPlugins.exposureBlend</li>
    <li>gimpPlugins.texturize</li>
    <li>glimpsePlugins.exposureBlend</li>
    <li>glimpsePlugins.texturize</li>
    <li>hhvm</li>
    <li>libsForQt512.ark</li>
    <li>libsForQt512.baloo</li>
    <li>libsForQt512.baloo-widgets</li>
    <li>libsForQt512.dolphin</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>168 packages skipped due to time constraints:</summary>
  <ul>
    <li>adminer</li>
    <li>ahoviewer</li>
    <li>php (apacheHttpdPackages.php ,php74)</li>
    <li>arcanist</li>
    <li>bookstack</li>
    <li>calligra</li>
    <li>cntk</li>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>deadbeef</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>22 packages built successfully:</summary>
  <ul>
    <li>ebook_tools</li>
    <li>ideviceinstaller</li>
    <li>idevicerestore</li>
    <li>katago</li>
    <li>katagoCPU</li>
    <li>lib3mf</li>
    <li>libsigrok</li>
    <li>libsigrok-0-3-0</li>
    <li>libsigrok4dsl</li>
    <li>libsigrokdecode</li>
    <li>libzip</li>
    <li>nxpmicro-mfgtools</li>
    <li>php73Extensions.zip</li>
    <li>php74Extensions.zip</li>
    <li>php80Extensions.zip</li>
    <li>python38Packages.r2pipe</li>
    <li>python39Packages.r2pipe</li>
    <li>radare2</li>
    <li>runzip</li>
    <li>sc-im</li>
    <li>sigrok-cli</li>
    <li>tldr</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/libraries/libzip/default.nix:46:3:
```
   |
46 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@ofborg eval

File: pkgs/development/libraries/libzip/default.nix [hunk 3]
```diff
@@ -23,20 +30,24 @@ stdenv.mkDerivation rec {
...
 
-  outputs = [ "out" "dev" ];
+  outputs = [ "out" "dev" "man" ];
 
-  nativeBuildInputs = [ cmake perl ];
+  nativeBuildInputs = [ cmake perl groff ];
   propagatedBuildInputs = [ zlib ];
+  buildInputs = (lib.optional withLZMA xz)
+    ++ (lib.optional withBzip2 bzip2)
+    ++ (lib.optional withOpenssl openssl );
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libzip/default.nix:39]:
```suggestion
  buildInputs = lib.optionals withLZMA [ xz ]
    ++ lib.optionals withBzip2 [ bzip2 ]
    ++ lib.optionals withOpenssl [ openssl ];
```

**Pull request merged**